lib/drives/pages/drive_chat_page.dart

```diff
@@ -16,107 +17,118 @@ class DriveChatPage extends StatefulWidget {
+                  for (Message message in snapshot.data!) {
+                    Chat chat = chats.firstWhere((Chat chat) => chat.id == message.chatId);
+                    if (chat.messages!.contains(message)) {
+                      chat.messages!.remove(message);
+                    }
+                    chat.messages!.add(message);
+                  }
+                  List<Card> widgets = chats.map((chat) => _buildChatWidget(chat)).toList();
+                  return ListView.separated(
+                    itemCount: widgets.length,
+                    itemBuilder: (BuildContext context, int index) {
+                      return widgets[index];
+                    },
+                    separatorBuilder: (BuildContext context, int index) {
+                      return const SizedBox(height: 10);
+                    },
+                  );
+                } else {
+                  return const Center(
+                    child: CircularProgressIndicator(),
+                  );
+                }
+              },
+            )
+          : Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              crossAxisAlignment: CrossAxisAlignment.center,
+              children: <Widget>[
+                Image.asset('assets/chat_shrug.png', scale: 8),
+                const SizedBox(height: 16),
+                Text(
+                  S.of(context).pageChatEmptyTitle,
+                  style: Theme.of(context).textTheme.headline6,
+                ),
+                const SizedBox(height: 8),
+                Padding(
+                  padding: const EdgeInsets.symmetric(horizontal: 16),
+                  child: Text(
+                    S.of(context).pageDriveChatEmptyMessage,
+                    textAlign: TextAlign.center,
+                  ),
+                ),
+              ],
+            ),
     );
   }
 
-  Widget _riderList(Set<Profile> riders) {
-    Widget ridersColumn = Container();
-    if (riders.isNotEmpty) {
-      ridersColumn = Column(
-        children: List.generate(
-          riders.length,
-          (index) => Padding(
-            padding: const EdgeInsets.symmetric(horizontal: 5.0, vertical: 5.0),
-            child: ProfileWidget(
-              riders.elementAt(index),
-              //TODO go to chat
-              onTap: () {},
-              actionWidget: const Icon(
-                Icons.chat,
-                color: Colors.black,
-                size: 36.0,
-              ),
-            ),
-          ),
+  Card _buildChatWidget(Chat chat) {
+    chat.messages!.sort((Message a, Message b) => b.createdAt!.compareTo(a.createdAt!));
+    Message? lastMessage = chat.messages!.isEmpty ? null : chat.messages!.first;
+    return Card(
+      child: InkWell(
+        child: ListTile(
+          leading: Avatar(chat.rider!),
+          title: Text(chat.rider!.username),
+          subtitle: lastMessage == null ? null : Text(lastMessage.content, maxLines: 1),
```

Comment:
this should also have the read/unread indicator imo -> that would also tell you whether the message was from you or from the other person.

Reply:
there is one just one line below, or what do you mean?

Reply:
No that's the number of unread messages... If I sent the last message, then I want to know at a glance whether the other person read it or not

Reply:
okay now i know what you mean i'll change that